File: HandSock-Server/src/main/java/com/server/handsock/ServerRunner.java
package com.server.handsock;

import com.server.handsock.console.ConsolePrints;
import org.springframework.stereotype.Component;
import com.corundumstudio.socketio.SocketIOServer;
import org.springframework.boot.CommandLineRunner;

@Component
public class ServerRunner implements CommandLineRunner {

    private final SocketIOServer socketIOServer;
    private final ConsolePrints consolePrints = new ConsolePrints();

    public ServerRunner(SocketIOServer socketIOServer) {
        this.socketIOServer = socketIOServer;
    }

    @Override
    public void run(String... args) {
        try {
            socketIOServer.start();
            consolePrints.printSuccessLog("Socket.IO server started!");
        } catch (Exception e) {
            consolePrints.printErrorLog(e);
        }
    }
}

File: HandSock-Server/src/main/java/com/server/handsock/ServerRunner.kt
package com.server.handsock

import com.corundumstudio.socketio.SocketIOServer
import com.server.handsock.common.utils.ConsoleUtils
import org.springframework.boot.CommandLineRunner
import org.springframework.stereotype.Component

@Component
class ServerRunner(
    private val socketIOServer: SocketIOServer
) : CommandLineRunner {
    override fun run(vararg args: String) {
        try {
            socketIOServer.start()
            ConsoleUtils.printSuccessLog("Socket.IO server started!")
        } catch (e: Exception) {
            ConsoleUtils.printErrorLog(e)
        }
    }
}